test/cljs/replumb/repl_test.cljs

```diff
@@ -189,323 +177,315 @@ dissoc!
+    (is (= "3" out) "(require '[foo.bar.baz :refer [referred-a]]) from foo.bar should retrieve the interned var value")))
+
+;; https://github.com/Lambda-X/replumb/issues/117
+(h/read-eval-call-test (assoc e/*target-opts* :load-fn! load/fake-load-fn!)
+  ["(require '[cljs.tools.reader :as r])"
+   "(binding [r/resolve-symbol (constantly 'cljs.user/x)] (r/read-string \"`x\"))"]
+  (let [out (unwrap-result @_res_)]
+    (is (success? @_res_) "Test for issues #117 should succeed")
+    (is (valid-eval-result? out) "Test for issues #117 should be a valid result")
+    (is (= "(quote cljs.user/x)" out) "Test for issues #117 should return (quote cljs.user/x)")))
+
+;; was require-node-test/require-when-read-file-return-nil
+(h/read-eval-call-test (assoc e/*target-opts* :load-fn! load/no-resource-load-fn!)
+  ["(require 'clojure.string)"
+   "(source clojure.string/trim)"
+   :after (repl/purge-cljs-user! '[clojure.string])]
+  (let [out (unwrap-result @_res_)]
+    (is (success? @_res_) "(source clojure.string/trim) should succeed.")
+    (is (valid-eval-result? out) "(source clojure.string/trim) should be a valid result")
+    (is (= "nil" out) "(source clojure.string/trim) should return nil")))
+
+;; was require-node-test/load-file-when-read-file-retuns-nil
+;; AR - This test loads the file anyways but it shouldn't !
+;; (h/read-eval-call-test (assoc e/*target-opts* :load-fn! load/no-resource-load-fn!)
+;;   ["(load-file \"foo/load.clj\")"]
+;;   (let [result (unwrap-result @_res_)]
+;;     (is (success? @_res_) "(load-file \"foo/load.clj\") should succeed")
+;;     (is (valid-eval-result? result) "(load-file \"foo/load.clj\") be a valid result")
+;;     (is (= "nil" result) "(load-file \"foo/load.clj\") should return nil")
+;;     (is (= (repl/current-ns) 'cljs.user) "(load-file \"foo/load.clj\") should not change namespace"))
+;;   (_reset!_ '[foo.load]))
+
+;; was source-node-test/source-corner-cases
+(h/read-eval-call-test (assoc e/*target-opts* :load-fn! load/no-resource-load-fn!)
+  ["(require 'clojure.string)"
+   "(source clojure.string/trim)"
+   :after (repl/purge-cljs-user! '[clojure.string])]
+  (let [source-string (unwrap-result @_res_)]
+    (println "ARSARSARSASR")
+    (is (success? @_res_) (str _msg_ "when *load-fn* returns nil should succeed."))
+    (is (valid-eval-result? source-string) (str _msg_ "when *load-fn* returns nil should be a valid result"))
+    (is (= "nil" source-string) (str _msg_ "when *load-fn* returns nil should return nil"))))
+
+(h/read-eval-call-test (-> e/*target-opts*
+                           (dissoc :load-fn!)
+                           (assoc :read-file-fn! nil))
+  ["(require 'clojure.string)"
+   "(source clojure.string/trim)"
+   :after (repl/purge-cljs-user! '[clojure.string])]
+  (let [source-string (unwrap-result @_res_)]
+    (println "--------------------\nARSARSARSASR\n")
```

Comment:
Intense debugging are you checking the unified already?

On March 15, 2016 4:34:41 PM PDT, Tomek <notifications@github.com> wrote:
>> +   "(source clojure.string/trim)"
>> +   :after (repl/purge-cljs-user! '[clojure.string])]
>> +  (let [source-string (unwrap-result @_res_)]
>> +    (println "ARSARSARSASR")
>> +    (is (success? @_res_) (str _msg_ "when *load-fn* returns nil
>should succeed."))
>> +    (is (valid-eval-result? source-string) (str _msg_ "when
>*load-fn* returns nil should be a valid result"))
>> +    (is (= "nil" source-string) (str _msg_ "when *load-fn* returns
>nil should return nil"))))
>> +
>> +(h/read-eval-call-test (-> e/*target-opts*
>> +                           (dissoc :load-fn!)
>> +                           (assoc :read-file-fn! nil))
>> +  ["(require 'clojure.string)"
>> +   "(source clojure.string/trim)"
>> +   :after (repl/purge-cljs-user! '[clojure.string])]
>> +  (let [source-string (unwrap-result @_res_)]
>> +    (println "--------------------\nARSARSARSASR\n")
>
>As above.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/Lambda-X/replumb/pull/162/files/2cff458e1a1e8fa3d0707cb60960164c3af940dc#r56259831

-ar